### PR TITLE
Add Limit Orders Support from Manifest

### PR DESCRIPTION
**Pull request by DonDuala**
# Pull Request Description
Add support for Agents to be able to create limit orders on Manifest. Implements create market and place limit order. Can consider adding cancel and other features in follow ups.

## Changes Made
This PR adds the following changes:
- support for manifest limit orders and market creation
  
## Implementation Details
- use manifest-sdk

## Transaction executed by agent 

Example transaction: 
Create Market: https://solscan.io/tx/59HgraXSzpmNtDovUfBUErjBwc2ggLtwfaYmrXdwR88HES6pLWHQWobHR85WmR3gHMavoEKLshCyJqx3Y6BZu8ku
Place limit order: https://solscan.io/tx/41d2CGSGdqkxXbuBDJwBniN3E9tN6hW7Tmo7oHQ1N1p5szejYhugKTLHaMNwS6KVp3DhMcTmZpkJMoWwKEpJAuf5

## Prompt Used

```
```

## Checklist
- [x] I have tested these changes locally
- [x] I have updated the documentation
- [x] I have added a transaction link
- [x] I have added the prompt used to test it

### CODE REVIEW

File: src/langchain/index.ts [hunk 1]
```diff
@@ -1351,6 +1433,7 @@ export function createSolanaTools(solanaKit: SolanaAgentKit) {
     new SolanaRaydiumCreateClmm(solanaKit),
     new SolanaRaydiumCreateCpmm(solanaKit),
     new SolanaOpenbookCreateMarket(solanaKit),
+    new SolanaManifestCreateMarket(solanaKit),
```
Comment (arihantbansal) [inline src/langchain/index.ts:1436]: Please add limit order tool as well

File: yarn.lock
Comment (arihantbansal) [inline yarn.lock:1]: Please remove this file / add to .gitignore. 

We prefer using pnpm

**Comment by thearyanag** [comment]
hey @DonDuala - please also enter the prompt you used to test It, and the transaction link from that interaction

**Comment by DonDuala** [comment]
> hey @DonDuala - please also enter the prompt you used to test It, and the transaction link from that interaction

image
https://solscan.io/tx/59HgraXSzpmNtDovUfBUErjBwc2ggLtwfaYmrXdwR88HES6pLWHQWobHR85WmR3gHMavoEKLshCyJqx3Y6BZu8ku

**Comment by DonDuala** [comment]
Limit order prompt and tx
image
https://solscan.io/tx/41d2CGSGdqkxXbuBDJwBniN3E9tN6hW7Tmo7oHQ1N1p5szejYhugKTLHaMNwS6KVp3DhMcTmZpkJMoWwKEpJAuf5

**Comment by DonDuala** [comment]
> Limit order prompt and tx image https://solscan.io/tx/41d2CGSGdqkxXbuBDJwBniN3E9tN6hW7Tmo7oHQ1N1p5szejYhugKTLHaMNwS6KVp3DhMcTmZpkJMoWwKEpJAuf5

Can view the market here btw https://app.manifest.trade/market/2Uj8277fkaVBtTU6Wp2GPRbQC86SkSdgQ2mp1Q5N2LHc

**Comment by thearyanag** [comment]
@DonDuala looks good

**Comment by DonDuala** [comment]
Added cancel and withdraw

Cancel: https://solscan.io/tx/4pRuXVpMzaTC9vARDk45ZuLcJL8vRyAsBV7WUUxXUTePUzaWjpqgVRmCy2KCgFPCqCFaaHUGJ3wAdxCrDqZ1ME5i
image


Withdraw: https://solscan.io/tx/4pLGEXZnQWR1cZVDapTefAcQvFUvfxqLYtWXK8ewvhibDWM8QaNyXwKcgGfqPu93GBsGcuLW71pFma9F2eeW6AB5
image

**Comment by DonDuala** [comment]
@thearyanag @arihantbansal Think this is good to merge

**Pull request merged**